sdv/multi_table/base.py

```diff
@@ -104,6 +104,11 @@ def set_address_columns(self, table_name, column_names, anonymization_level='ful
             anonymization_level (str):
                 The anonymization level to use for the address transformer.
         """
+        if not self._table_synthesizers.get(table_name):
+            raise ValueError(
+                f"Unknown table name '{table_name}'. Please choose a table name from the metadata."
+            )
```

Comment:
I think you could reuse the `_validate_table_name` method for this. Also, I think that method should be changed to raise a `ValueError` like you do here.